README.md

```diff
@@ -4,222 +4,1008 @@
+```
+
+So we can see that the help text matches the structure of our `DefaultCommand` struct.
+
+Next, let's try out our command!
+
+```bash
+# Even number
+$> ./mytool 60
+Program exited with status code 1
+
+# Odd number
+$> ./mytool 59
+Program exited with status code 0
+
+# No number (Error message doesn't look great yet)
+$> ./mytool
+ERROR: The following required positional arguments were not provided: ["[0] number"]
+Program exited with status code -1
+
+# Too many numbers
+$> ./mytool 1 2
+ERROR: Stray positional arg found: '2'
+Program exited with status code -1
+```
+
+Excellent, we can see that with little to no work, our command performs as expected while rejecting invalid use cases.
+
+## Named arguments
+
+Right, so what if the user decides that they want us to return `0` instead of `1`, and `1` instead of `0`. In other words, reverse the output?
+
+Well, let's make them pass in a named argument called `--mode`, which maps directly to an `enum` inside of our D code, to select what behavior we want. Did I forget to mention JCLI can do that?
+
+```d
+enum Mode
+{
+    normal,  // Even returns 1. Odd returns 0.
+    reversed // Even returns 0. Odd return 1.
+}
+
+@Command(null, "The default command.")
+struct DefaultCommand
+{
+    @CommandPositionalArg(0, "number", "The number to check.")
+    int number;
+
+    @CommandNamedArg("mode", "Which mode to use.")
+    Mode mode;
+
+    int onExecute()
+    {
+        if(this.mode == Mode.normal)
+            return number % 2 == 0 ? 1 : 0;
+        else
+            return number % 2;
+    }
+}
+```
+
+To start off, we create an `enum` called `Mode`, and give the members `normal` and `reversed`. JCLI knows how to map strings into enums.
+
+Next, inside of `DefaultCommand` we create a member field called `Mode mode;` that is decorated with the `CommandNamedArg` UDA.
+
+The first parameter is the name of the argument, which is actually important this time as this determines what name the user has to use.
+
+The second parameter is just the description.
+
+Then inside of `onExecute` we just check what `mode` was set to and do stuff based off of its value.
+
+Time to test! Remember to run `dub build` first.
+
+Let's have a quick look at the help text first, to see the changes being reflected:
+
+```bash
+$> ./mytool --help
+Usage: DEFAULT {0/number} [mode]
+
+Description:
+    The default command.
+
+Positional Args:
+    0,number                     - The number to check.
+
+Named Args:
+    --mode                       - Which mode to use.
+```
+
+And now let's test our functionality:
+
+```bash
+# JCLI supports most common argument styles.
+
+# Even (Normal)
+$> mytool 60 --mode normal
+Program exited with status code 1
+
+# Even (Reversed)
+$> mytool 60 --mode=reversed
+Program exited with status code 0
+
+# Bad value for mode
+$> mytool 60 --mode lol
+std.conv.ConvException@\src\phobos\std\conv.d(2817): Mode does not have a member named 'reverse'
```

Comment:
First of all, error text mentions `reverse` but `lol` is passed.

Error text should be better:
```suggestion
Invalid value for argument '--mode': 'lol'
```

Reply:
That's mostly down to laziness. Opened #10 for it though.